modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -4045,9 +4060,61 @@ private void assignPartitionsStates() {
             throw new IgniteException("Failed to assign partition states", e);
         }
 
+        if (log.isInfoEnabled() && !F.isEmpty(supplayInfoMap))
+            printPartiotnRebalancingFully(supplayInfoMap);
+
         timeBag.finishGlobalStage("Assign partitions states");
     }
 
+    /**
+     * Prints detail information about partitions which did not have reservation
+     * history enough for historical rebalance.
+     *
+     * @param supplayInfoMap Map contains information about supplying partitions.
+     */
+    private void printPartiotnRebalancingFully(Map<String, List<SupplayPartitionInfo>> supplayInfoMap) {
+        if (hasPartitonToLog(supplayInfoMap, false)) {
+            log.info("Partitions do not have WAL reservation for historical rebalance: [" +
+                supplayInfoMap.entrySet().stream().map(entry ->
+                    "[grp=" + entry.getKey() + " part=[" + S.compact(entry.getValue().stream()
+                        .filter(info -> !info.isHistoryReserved())
+                        .map(info -> info.part()).collect(Collectors.toSet())) + "]]"
+                ).collect(Collectors.joining(", ")) + ']');
+        }
+
+        if (hasPartitonToLog(supplayInfoMap, true)) {
+            log.info("WAL reservation does not enough for historical rebalance: [" +
```

Comment:
"Partitions were reserved, but maximum available counter is greater than demanded"

Reply:
Done.